src/Tools/dotnet-trace/CommandLine/Commands/MonitorCommand.cs

```diff
@@ -0,0 +1,267 @@
+                    {
+                        ProcessLauncher.Launcher.ChildProc.Kill();
+                    }
+                    return ErrorCodes.SessionCreationError;
+                }
+            }
+            else
+            {
+                diagnosticsClient = new DiagnosticsClient(processId);
+            }
+
+            EventPipeSession session = diagnosticsClient.StartEventPipeSession(providerCollection);
+            if (ProcessLauncher.Launcher.HasChildProc)
+            {
+                diagnosticsClient.ResumeRuntime();
+            }
+
+            ManualResetEvent shouldExit = new ManualResetEvent(false);
+            ct.Register(() => shouldExit.Set());
+
+            Task parserTask = Task.Run(() =>
+            {
+                try
+                {
+                    using (EventPipeEventSource source = new EventPipeEventSource(session.EventStream))
+                    {
+
+                        foreach (IDiagnosticProfileHandler handler in GetAllHandlers(providerCollection))
+                        {
+                            handler.RegisterHandler(source);
+                        }
+                        source.Process();
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Debug.WriteLine(ex.ToString());
+                }
+                finally
+                {
+                    shouldExit.Set();
+                }
+            });
+
+            while(!shouldExit.WaitOne(250))
+            {
+                while (true)
+                {
+                    if (shouldExit.WaitOne(250))
+                    {
+                        StopSession(session);
+                        return 0;
+                    }
+                    if (Console.KeyAvailable)
+                    {
+                        break;
+                    }
+                }
+                ConsoleKey cmd = Console.ReadKey(true).Key;
+                if (cmd == ConsoleKey.Q)
+                {
+                    StopSession(session);
+                    break;
+                }
+            }
+
+            return await Task.FromResult(1);
+        }
+
+        /// <summary>
+        /// Small wrapper around EventPipeSession.Stop() to catch any Timeout exceptions
+        /// A TimeoutException or ServerNotAvailableException may be thrown while stopping the session in the case of:
+        ///     1. The app exited before the user stopped the tool.
+        ///     2. The app exited after the user stopped the tool, but before the tool sent stop command.
+        /// </summary>
+        /// <param name="session"></param>
+        private static void StopSession(EventPipeSession session)
+        {
+            try
+            {
+                session.Stop();
+            }
+            catch (TimeoutException) { }
+            catch (ServerNotAvailableException) { }
```

Comment:
Some of the existing code also handles EndOfStreamException and PlatformNotSupportedException. Are these no longer necessary?